src/cpp-utils/thread/debugging_nonwindows.cpp

```diff
@@ -5,6 +5,14 @@
 #include <thread>
 #include <pthread.h>
 #include <cpp-utils/assert/assert.h>
+#if !(defined(__GLIBC__) || defined(__APPLE__))
```

Comment:
Was under the impression MacOS implemented its own `pthread_getname_np`. Should this be `|| !defined(APPLE)`?

Reply:
The parentheses may make it a little non-obvious: that says !    (GLIBC || APPLE), meaning "not glibc nor Apple".